components/sst_importer/src/sst_importer.rs

```diff
@@ -231,26 +235,58 @@ impl SSTImporter {
             Bound::Excluded(_) => unreachable!(),
         };
         while iter.valid() {
-            let old_key = keys::origin_key(iter.key());
-            if is_after_end_bound(old_key, &range_end) {
+            let old_key = {
+                if rewrite_rule.new_timestamp != 0 {
+                    Cow::from(
```

Comment:
Is `Cow` necessary?  Seems the `old_key` would never be write.

Reply:
it is used to unify `Vec<u8>` and `&[u8]`, or it needs extra memory allocation for `&[u8]`

Reply:
If I were to write this code I'd make it like this:

```rust
        let is_write_cf = meta.get_cf_name() == "write";
        let new_ts = TimeStamp::new(rewrite_rule.new_timestamp);

        while iter.valid() {
            // Don't rewrite anything while we're still checking if old_key is valid.
            let old_key = keys::origin_key(iter.key());
            if is_after_end_bound(old_key, &range_end) {
                break;
            }
            if !old_key.starts_with(old_prefix) {
                return Err(Error::WrongKeyPrefix(
                    "Key in SST",
                    old_key.to_vec(),
                    old_prefix.to_vec(),
                ));
            }

            key.truncate(new_prefix_data_key_len);
            key.extend_from_slice(&old_key[old_prefix.len()..]);
            let mut value = Cow::Borrowed(iter.value());

            // rewrite the _new_ key.
            if !new_ts.is_zero() {
                // consider introducing `Key::rewrite_ts_for(&mut [u8], TimeStamp)` 
                // so we don't need to adjust the length but directly rewrite the last 8 bytes
                key = Key::from_encoded(key)
                    .truncate_ts()
                    .map(|k| k.append_ts(new_ts))
                    .map_err(|e| {
                        Error::BadFormat(format!("key {}: {}", hex::encode_upper(old_key)), e)
                    })?
                    .into_encoded();
                if is_write_cf {
                    let mut write = WriteRef::parse(value).map_err(|e| {
                        Error::BadFormat(format!("value {}: {}", hex::encode_upper(value), e))
                    })?;
                    write.start_ts = new_ts;
                    value = Cow::Owned(write.to_bytes());
                }
            }

            sst_writer.put(&key, &value)?;
            iter.next();
            if first_key.is_none() {
                first_key = Some(keys::origin_key(&key).to_vec());
            }
        }
```